pkg/utils/file.go

```diff
@@ -64,3 +67,24 @@ func GetFileSize(file string) (int64, error) {
 	}
 	return stat.Size(), nil
 }
+
+// WriteFileAtomic writes file to temp and atomically move when everything else succeeds.
+func WriteFileAtomic(filename string, data []byte, perm os.FileMode) error {
+	dir, name := path.Dir(filename), path.Base(filename)
+	f, err := ioutil.TempFile(dir, name)
+	if err != nil {
+		return err
+	}
+	n, err := f.Write(data)
+	f.Close()
+	if err == nil && n < len(data) {
+		err = io.ErrShortWrite
+	} else {
+		err = os.Chmod(f.Name(), perm)
+	}
+	if err != nil {
+		os.Remove(f.Name())
```

Comment:
what if `os.Remove` call fails here, do we need to output a log?

Reply:
https://github.com/pingcap/dm/pull/1615/commits/feab697c8a39b13e80ca8eaac0a0b3c8265105a7